src/main/java/com/minecolonies/coremod/client/gui/AbstractWindowRequestTree.java

```diff
@@ -353,6 +386,58 @@ public void updateElement(final int index, final Pane rowPane)
         });
     }
 
+    /**
+     * Checks if the request is fulfillable
+     *
+     * @param tRequest the request to check if it's fulfillable
+     */
+    public Boolean fulfillable(final IRequest tRequest)
```

Comment:
This is still not complete in the request tree class there are a bunch more conditions that have to be met to hide this sometimes:

Those are 5 conditions to hide fullfill and 1 to hide cancel. That needs to be extracted into a method and also called in here.

```
if (wrapper.overruleable && canFulFill())
                {
                    if (wrapper.getDepth() > 0)
                    {
                        if (!(request.getRequester() instanceof IBuildingBasedRequester)
                              || !((IBuildingBasedRequester) request.getRequester())
                                    .getBuilding(colony.getRequestManager(),
                                      request.getId()).map(
                            iRequester -> iRequester.getLocation()
                                            .equals(building.getLocation())).isPresent())
                        {
                            rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
                        }
                        else
                        {
                            request.getRequestOfType(IDeliverable.class).ifPresent((IDeliverable requestRequest) -> {
                                if (!isCreative && !InventoryUtils.hasItemInItemHandler(new InvWrapper(inventory), requestRequest::matches))
                                {
                                    rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
                                }
                            });

                            if (!(request.getRequest() instanceof IDeliverable))
                            {
                                rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
                            }
                        }
                        rowPane.findPaneOfTypeByID(REQUEST_CANCEL, ButtonImage.class).hide();
                    }
                    else
                    {
                        request.getRequestOfType(IDeliverable.class).ifPresent((IDeliverable requestRequest) -> {
                            if (!isCreative && !InventoryUtils.hasItemInItemHandler(new InvWrapper(inventory), requestRequest::matches))
                            {
                                rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
                            }
                        });
                    }
                }
                else
                {
                    rowPane.findPaneOfTypeByID(REQUEST_FULLFIL, ButtonImage.class).hide();
                }
```